### PR TITLE
fix(rhc): fix bug in healthcheck dockerfile

### LINKED ISSUE #2351: Bug: replica-healthcheck docker image 1.0.1 fails
**Describe the bug**

replica-healthcheck 1.0.1 started from official docker image fails with

```
replica-healthcheck_1   | npm ERR! A complete log of this run can be found in:
replica-healthcheck_1   | npm ERR!     /root/.npm/_logs/2022-03-22T14_08_43_686Z-debug.log
replica-healthcheck_1   | npm ERR! code ENOENT
replica-healthcheck_1   | npm ERR! syscall open
replica-healthcheck_1   | npm ERR! path /opts/optimism/packages/replica-healthcheck/package.json
replica-healthcheck_1   | npm ERR! errno -2
replica-healthcheck_1   | npm ERR! enoent ENOENT: no such file or directory, open '/opts/optimism/packages/replica-healthcheck/package.json'
replica-healthcheck_1   | npm ERR! enoent This is related to npm not being able to find a file.
replica-healthcheck_1   | npm ERR! enoent
```

**Expected behavior**

Can an enhancement request be added? replica-healthcheck docker images have been unhealthy for a while. How about adding CI/CD that specifically tests the docker build pipeline for it?

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Fixes a bug in the replica-healthcheck dockerfile. Also includes
replica-healthcheck as part of the ops setup by default to make this
stuff easier to catch and debug in the future. We will likely still want
to add integration tests that query the metrics defined in the
replica-healthcheck to confirm that everything is working as expected.

Fixes #2351

### CODE REVIEW

File: ops/docker-compose.yml [hunk 1]
```diff
@@ -197,6 +197,25 @@ services:
...
+    build:
+      context: ..
+      dockerfile: ./ops/docker/Dockerfile.packages
+      target: replica-healthcheck
+    image: ethereumoptimism/replica-healthcheck:${DOCKER_TAG_REPLICA_HEALTHCHECK:-latest}
+    entrypoint: ./healthcheck.sh
+    environment:
+      HEALTHCHECK__REFERENCE_RPC_PROVIDER: http://l2geth:8545
+      HEALTHCHECK__TARGET_RPC_PROVIDER: http://replica:8545
+      RETRIES: 60
```
[the remark this answers is not in the document]
Comment (smartcontracts) [reply ops/docker-compose.yml:215]: Sure

File: ops/docker-compose.yml [hunk 2]
```diff
@@ -197,6 +197,25 @@ services:
       - ${REPLICA_HTTP_PORT:-8549}:8545
       - ${REPLICA_WS_PORT:-8550}:8546
 
+  replica-healthcheck:
+    depends_on:
+      - l2geth
+      - replica
+    deploy:
+      replicas: 1
```
Comment (tynes) [inline ops/docker-compose.yml:205]: I have a preference to have `replicas: 0` by default, we don't need this to run during local development, the setup is already becoming beefy

File: packages/common-ts/src/base-service/base-service-v2.ts [hunk 3]
```diff
@@ -203,6 +229,11 @@ export abstract class BaseServiceV2<
       return acc
     }, {}) as TMetrics
 
+    // Create the metrics server.
+    this.metricsRegistry = prometheus.register
```
Comment (smartcontracts) [inline packages/common-ts/src/base-service/base-service-v2.ts:233]: Whoops. I forgot to expose the metrics at all.

File: ops/docker-compose.yml (hunk 2 above)
Comment (smartcontracts) [reply ops/docker-compose.yml]: Ok, done. Not included by default but will run in integration tests.

File: ops/docker-compose.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (smartcontracts) [reply ops/docker-compose.yml]: Done, no retry scripts, caught cleanly in healthcheck server.

**Review by optimisticben — APPROVED** [review]
LGTM

**Pull request merged**